api/src/main/java/ru/urururu/sanity/api/InstructionParser.java

```diff
@@ -0,0 +1,158 @@
+package ru.urururu.sanity.api;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import ru.urururu.sanity.api.cfg.*;
+import ru.urururu.util.FinalMap;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public abstract class InstructionParser<T, V, I, B, Ctx extends CfgBuildingCtx<T, V, I, B, Ctx>> {
+    @Autowired
+    protected ParsersFacade<T, V, I, B, Ctx> parsers;
+
+    public Cfe parse(Ctx ctx, I instruction) {
+        try {
+            return doParse(ctx, instruction);
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)